### PR TITLE
Feature/meshaxis

### PR DESCRIPTION
**Pull request by martinfalk**
Fixes:
* incorrect TetraMesh bounding box in `TetraMeshBoundingBox`

Changes proposed in this PR:
 * new processor `MeshAxis` similar to `VolumeAxis`
 * refactored VolumeAxis processor into commonly used components and properties (`Axis3DProcessorHelper`)

This has been tested on: Windows, MSVC 17.8.5

### CODE REVIEW

File: modules/plottinggl/src/utils/axis3dprocessorhelper.cpp [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        offset = entity->getOffset();
+
+        worldTrafo = entity->getCoordinateTransformer().getDataToWorldMatrix();
+    }
+
+    std::string xCaption{"x"};
+    std::string yCaption{"y"};
+    std::string zCaption{"z"};
+
+    if (auto volume = dynamic_cast<const Volume*>(entity)) {
```
Comment (petersteneteg) [inline modules/plottinggl/src/utils/axis3dprocessorhelper.cpp]: you could add an extra argument for a std::function<std::string(size_t axis)> to get the captions, then you don'e have to deal with the volume and dynamic cast it here...

Comment (petersteneteg) [reply modules/plottinggl/src/utils/axis3dprocessorhelper.cpp]: Or we go all the way and add that to SpatialEntity as we talked about

**Review by petersteneteg — DISMISSED** [review]
might fix the range warning and remove 
include/inviwo/core/datastructures/image/imageramprecision.h
if you want to...

File: modules/plottinggl/src/utils/axis3dprocessorhelper.cpp (hunk 1 above)
Comment (martinfalk) [reply modules/plottinggl/src/utils/axis3dprocessorhelper.cpp]: Added virtual function `const Axis* getAxis(size_t) const` to SpatialEntity.

Comment (petersteneteg) [reply modules/plottinggl/src/utils/axis3dprocessorhelper.cpp]: hmm you still need the dims? 
should cast to StructuredGridEntity in principle... that should also pull in the layer I suppose

File: modules/tetramesh/src/datastructures/volumetetramesh.cpp [hunk 2]
```diff
@@ -171,4 +171,11 @@ dvec2 VolumeTetraMesh::getDataRange() const {
     }
 }
 
+const Axis* VolumeTetraMesh::getAxis(size_t index) const {
+    if (volume_) {
+        return volume_->getAxis(index);
```
Comment (petersteneteg) [inline modules/tetramesh/src/datastructures/volumetetramesh.cpp:176]: this would have been difficult the just the axis member in the base...

**Review by petersteneteg — DISMISSED** [review]

File: modules/plottinggl/src/utils/axis3dprocessorhelper.cpp (hunk 1 above)
Comment (martinfalk) [reply modules/plottinggl/src/utils/axis3dprocessorhelper.cpp]: > hmm you still need the dims?

Yes, for referring to texels/voxels.
Good point with `StructuredGridEntity`.

**Review by petersteneteg — APPROVED** [review]

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**